exe/wallet/Main.hs

```diff
@@ -411,7 +437,54 @@ execHttpBridge args _ = do
     nw <- HttpBridge.newNetworkLayer @n (getPort bridgePort)
     waitForConnection nw defaultRetryPolicy
     let tl = HttpBridge.newTransactionLayer @n
-    wallet <- newWalletLayer @_ @(HttpBridge n) tracer block0 byronFeePolicy db nw tl
+    wallet <- newWalletLayer tracer HttpBridge.block0 byronFeePolicy db nw tl
+    Server.withListeningSocket walletListen $ \(port, socket) -> do
+        tracerIPC <- appendName "DaedalusIPC" tracer
+        tracerApi <- appendName "api" tracer
+        let beforeMainLoop = logInfo tracer $
+                "Wallet backend server listening on: " <> toText port
+        let settings = Warp.defaultSettings
+                & setBeforeMainLoop beforeMainLoop
+        let ipcServer = daedalusIPC tracerIPC port
+        let apiServer = Server.start settings tracerApi socket wallet
+        race_ ipcServer apiServer
+
+-- Instantiate a new 'NetworkLayer' for 'Jormungandr'.
+-- Returns the genesis block and fee policy for starting a 'WalletLayer'.
+newNetworkLayer
+    :: BaseUrl
+    -> Hash "Genesis"
+    -> IO (NetworkLayer (Jormungandr n) IO, Block Tx, FeePolicy)
+newNetworkLayer url block0H = do
+    mgr <- newManager defaultManagerSettings
+    let jormungandr = Jormungandr.mkJormungandrLayer mgr url
+    let nl = Jormungandr.mkNetworkLayer jormungandr
+    waitForConnection nl defaultRetryPolicy
+    block0 <- unsafeRunExceptT $ getBlock jormungandr (coerce block0H)
```

Comment:
FYI, we *can* retrieve `block0H` ourselves

```haskell
{-# LANGUAGE OverloadedStrings #-}
import Cardano.Wallet.Jormungandr.Network
import Cardano.Wallet.Primitive.Types
import Cardano.Wallet.Unsafe
import Data.Coerce
    ( coerce )

main :: IO ()
main = do
    mgr <- newManager defaultManagerSettings
    let j = mkJormungandrLayer mgr (BaseUrl Http "localhost" 8081 "/api")
    let absentParent = Hash "\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL\NUL"
    [block0H] <- unsafeRunExceptT $ getDescendantIds j absentParent 1
    block0 <- unsafeRunExceptT $ getBlock j (coerce block0H)
    print block0
```

Reply:
Yes and no. We do want the user to provide it such that we can verify they match!